addons/test_mail/tests/test_mail_activity.py

```diff
@@ -759,3 +761,253 @@ def test_week_grouping(self):
+            })
+
+    def get_default_deadline(self, activity_type, context=None):
+        return self.env['mail.activity'].with_context(**(context or {}))._calculate_date_deadline(activity_type)
+
+    def check_plan_activities_created(self, plan, on_record,
+                                      sole_activities=False, given_responsible_id=None, given_date_deadline=None):
+        expected_number_of_activity = len(plan.template_ids)
+        activities = self.get_last_activities(on_record, None if sole_activities else expected_number_of_activity)
+        default_responsible_id = given_responsible_id or self.env.user
+        self.assertEqual(len(activities), expected_number_of_activity)
+        for activity, template in zip(activities, plan.template_ids):
+            self.assertEqual(activity.activity_type_id, template.activity_type_id)
+            self.assertEqual(activity.summary, template.summary)
+            self.assertEqual(activity.note, template.note)
+            responsible_id = default_responsible_id if template.responsible == 'on_demand' else template.responsible_id
+            self.assertEqual(activity.user_id, responsible_id)
+            self.assertEqual(activity.date_deadline,
+                             given_date_deadline or self.get_default_deadline(template.activity_type_id))
+
+    def check_plan_message_posted(self, plan, on_record, given_responsible_id=None, given_date_deadline=None):
+        default_responsible_id = given_responsible_id or self.env.user
+        message = self.get_last_message(on_record)
+        self.assertIn(f'The plan "{plan.name}" has been started', message.body)
+        for template in plan.template_ids:
+            self.assertIn(template.summary, message.body)
+            responsible_id = default_responsible_id if template.responsible == 'on_demand' else template.responsible_id
+            date_deadline = given_date_deadline or self.get_default_deadline(template.activity_type_id)
+            self.assertIn(f'{template.summary or template.activity_type_id.name}, '
+                          f'assigned to {responsible_id.name}, due on the {date_deadline}', message.body)
+
+    def check_wizard_plan(self, wizard, records, given_responsible_id=None, given_date_deadline=None):
+        action = wizard.action_schedule_plan()
+        self.check_action_feedback_schedule_plan(action, records)
+        for record in records:
+            self.check_plan_activities_created(self.plan_party, record, given_date_deadline=given_date_deadline,
+                                               given_responsible_id=given_responsible_id)
+            self.check_plan_message_posted(self.plan_party, record, given_date_deadline=given_date_deadline,
+                                           given_responsible_id=given_responsible_id)
+
+    @mute_logger('odoo.tests', 'odoo.addons.mail.models.mail_mail', 'odoo.models.unlink')
+    def test_activity_schedule(self):
```

Comment:
Would make tests on test records defined in 'test_mail', that way we don't depend on 'partner' specific implementation, fields, ... all tests in 'test_mail' are independent from business models (to be sure to test the feature and not the business model).